x/proof/keeper/msg_server_submit_proof.go

```diff
@@ -35,9 +36,16 @@ func init() {
 // A proof that's stored on-chain is what leads to rewards (i.e. inflation)
 // downstream, making the series of checks a critical part of the protocol.
 // TODO_BLOCKER: Prevent proof upserts after the tokenomics module has processes the respective session.
+// TODO_TECHDEBT_DISCUSS: Do we need to validate if the signature on the Proof
+// message corresponds to the supplier before upserting?
 func (k msgServer) SubmitProof(ctx context.Context, msg *types.MsgSubmitProof) (*types.MsgSubmitProofResponse, error) {
+	// TODO_BLOCKER_DISCUSS: A potential issue with doing proof validation inside
+	// `SubmitProof` is that we will not be storing false proofs on-chain (e.g. for slashing purposes).
+	// This could be considered a feature (e.g. less state bloat against sybil attacks)
+	// or a bug (i.e. no mechanisms for slashing suppliers who submit false proofs).
+	// Revisit this prior to mainnet launch as to whether the business logic for settling sessions should be in EndBlocker or here.
 	logger := k.Logger().With("method", "SubmitProof")
-	logger.Debug("about to start submitting proof")
+	logger.Info("About to start submitting proof")
```

Comment:
TL;DR, may I suggest we go with `logger.Info()` for now so that the logs are visible but also add a comment to replace with a debug level log if/when we swap out the cosmos-sdk logger for on-chain logging?

I have a strong opinion about these being debug level logs due to their nature; however, the cosmos-sdk logger's debug level is essentially useless as it includes a bunch of extra junk from within cosmos-sdk that drowns out everything else. I think we probably ought to use a different logger and/or add a polylog implementation that wraps cosmos-sdk logger and allows us to have an appchain-specific debug level.